core/src/main/java/org/elasticsearch/index/similarity/SimilarityService.java

```diff
@@ -63,6 +63,8 @@ public SimilarityService(IndexSettings indexSettings, Map<String, BiFunction<Str
         Map<String, Settings> similaritySettings = this.indexSettings.getSettings().getGroups(IndexModule.SIMILARITY_SETTINGS_PREFIX);
         for (Map.Entry<String, Settings> entry : similaritySettings.entrySet()) {
             String name = entry.getKey();
+            if(BUILT_IN.containsKey(name))
```

Comment:
I think this is tricky to do here without any backwards compatibility. Users were able to do that in the past so on upgrade they will get an exception and they will not be able to fix it. I think we have to move this into where we validate our settings such that it only applied to new indices and old indices should still have this ability? I think we need to add something to the settings infrastructure to prevent this from being added but we should fail here too if the index is onOrAfter 5.0 so this means we should make this:

```
 if(BUILT_IN.containsKey(name) && indexSettings.getIndexVersionCreated().onOrAfter(Version.V_3_0_0)) {
                throw new IllegalArgumentException("Cannot redefine built-in Similarity [" + name + "]");
}
```